### PR TITLE
Improve handling of images with spaces in file name 

### LINKED ISSUE #5969: Improve handling of images with spaces in file name
## Feature Description

WordPress by default replaces spaces in filenames with dashes upon upload.

However, some sites might still end up keeping the spaces in the filenames when using certain plugins.

In that case, the editor doesn't handle those cases very well, with images not being displayed properly due to the image filenames not being properly URL-encoded.

Two areas where I noticed this:

* The `srcset` attribute when displaying images in the editor
* The `srcset` attribute in the final output on the frontend

The latter also causes AMP validation issues for the story.

This could be solved by properly URL-encoding images in `srcset` before using.

## Alternatives Considered

This is quite an edge case that we could also consider not supporting

## Additional Context

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

### PR DESCRIPTION
**Pull request by spacedmonkey**
## Summary
Wrap  `encodeURI` around `source_url` in calculateSrcSet and add a unit test for it.

## Testing Instructions

- [x] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1.

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This code is covered by automated tests (unit, integration, and/or e2e) to verify it works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #5969

### CODE REVIEW

File: packages/media/src/test/calculateSrcSet.js [hunk 1]
```diff
@@ -117,6 +117,74 @@ describe('calculateSrcSet', () => {
     expect(srcSet).toBe('large-url 400w');
   });
 
+  it('should sizes with spaces', () => {
```
Comment (swissspidy) [inline packages/media/src/test/calculateSrcSet.js]:
```suggestion
  it('should encode URLs with spaces', () => {
```

File: packages/media/src/test/calculateSrcSet.js [hunk 2]
```diff
@@ -117,6 +117,74 @@ describe('calculateSrcSet', () => {
...
+      },
+    };
+
+    const srcSet = calculateSrcSet(resource);
+    expect(srcSet).toBe(
+      'large%2520url 300w,medium%2520url 200w,small%2520url 100w'
+    );
+  });
+
+  it('should sizes with multiple spaces', () => {
```
Comment (swissspidy) [inline packages/media/src/test/calculateSrcSet.js]:
```suggestion
  it('should encode URLs with multiple spaces', () => {
```

File: packages/media/src/test/calculateSrcSet.js [hunk 3]
```diff
@@ -117,6 +117,74 @@ describe('calculateSrcSet', () => {
...
+      },
+    };
+
+    const srcSet = calculateSrcSet(resource);
+    expect(srcSet).toBe(
+      'large%20%20%20%20%20%20url 300w,medium%20%20%20%20%20url 200w,small%20%20%20%20%20%20url 100w'
+    );
+  });
+
+  it('should sizes with empty source_url', () => {
```
Comment (swissspidy) [inline packages/media/src/test/calculateSrcSet.js]:
```suggestion
  it('should ignore sizes with empty URLs', () => {
```

File: packages/media/src/test/calculateSrcSet.js [hunk 4]
```diff
@@ -117,6 +117,74 @@ describe('calculateSrcSet', () => {
...
+        img2: { width: 200, height: 200, source_url: 'medium url' },
+        img3: { width: 300, height: 300, source_url: 'large url' },
+      },
+    };
+
+    const srcSet = calculateSrcSet(resource);
+    expect(srcSet).toBe('large%20url 300w,medium%20url 200w,small%20url 100w');
+  });
+
+  it('should sizes with encoded spaces', () => {
```
Comment (swissspidy) [inline packages/media/src/test/calculateSrcSet.js]:
```suggestion
  it('should encode already encoded URLs', () => {
```

Although this test basically just tests `encodeURI` itself, so seems a bit overkill

**Review by swissspidy — APPROVED** [review]
Just a few nits

**Comment by spacedmonkey** [comment]
Addressed feedback.

**Review by swissspidy — APPROVED** [review]

**Pull request merged**